### PR TITLE
Refactoring Azure resources in main.tf, variables.tf

### PR DESCRIPTION
**Pull request by Simranpal**
With this PR, the remaining resources (mainly provider, resource group & network) are moved into the main.tf, infrastructure.tf.
Also updated readme to reflect the changes. 
This PR replaces https://github.com/SUSE/ha-sap-terraform-deployments/pull/343

### CODE REVIEW

File: azure/main.tf [hunk 1]
```diff
@@ -117,35 +116,26 @@ module "hana_node" {
...
-  monitoring_public_version   = var.monitoring_public_version
-  monitoring_srv_ip           = var.monitoring_srv_ip
-  public_key_location         = var.public_key_location
-  private_key_location        = var.private_key_location
-  admin_user                  = var.admin_user
-  host_ips                    = var.host_ips
-  reg_code                    = var.reg_code
-  reg_email                   = var.reg_email
-  reg_additional_modules      = var.reg_additional_modules
-  additional_packages         = var.additional_packages
```
Comment (arbulu89) [inline azure/main.tf]: Why have you removed so many variables here? I would say there are needed, right?

File: azure/main.tf [hunk 2]
```diff
@@ -156,11 +146,6 @@ module "iscsi_server" {
   network_subnet_id      = azurerm_subnet.mysubnet.id
   sec_group_id           = azurerm_network_security_group.mysecgroup.id
   storage_account        = azurerm_storage_account.mytfstorageacc.primary_blob_endpoint
-  iscsi_srv_uri          = var.iscsi_srv_uri
```
Comment (arbulu89) [inline azure/main.tf]: Why have you removed these variables?

File: azure/main.tf [hunk 3]
```diff
@@ -106,7 +106,6 @@ module "hana_node" {
   reg_code                      = var.reg_code
   reg_email                     = var.reg_email
   reg_additional_modules        = var.reg_additional_modules
-  additional_packages           = var.additional_packages
```
Comment (arbulu89) [inline azure/main.tf]: Why have you removed this variable?

**Comment by Simranpal** [comment]
Hi @arbulu89 the variables in this PR were missing, because I merged the other PR https://github.com/SUSE/ha-sap-terraform-deployments/pull/352 and this PR was not updated with the changes.

**Comment by Simranpal** [comment]
Thanks @arbulu89 a terraform infrastructure module makes sense. I will do that along with next provider.

I tested with deployment now with hana and monitoring and works fine

**Pull request merged**